CachedExecutorServiceDelegateTest shutdownNow()

from https://hazelcast-l337.ci.cloudbees.com/job/new-lab-fast-pr/6849/testReport/junit/com.hazelcast.util.executor/CachedExecutorServiceDelegateTest/shutdownNow/

```
com.hazelcast.util.executor.CachedExecutorServiceDelegateTest.shutdownNow

Failing for the past 1 build (Since Failed#6849 )
Took 0.11 sec.
Error Message

expected:<1> but was:<0>
Stacktrace

java.lang.AssertionError: expected:<1> but was:<0>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.util.executor.CachedExecutorServiceDelegateTest.shutdownNow(CachedExecutorServiceDelegateTest.java:294)
```

It looks like a bug in a test - the "indefinite" task is parked, but it can be woke-up spuriously.

--- a/hazelcast/src/test/java/com/hazelcast/util/executor/CachedExecutorServiceDelegateTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/util/executor/CachedExecutorServiceDelegateTest.java
@@ -128,11 +128,14 @@ public class CachedExecutorServiceDelegateTest {
         int queueSize = 10;
         ManagedExecutorService executorService = newManagedExecutorService(1, queueSize);
 
-        startInfinitelyRunningTask(executorService);
-        executeNopTask(executorService);
-
-        assertEquals(1, executorService.getQueueSize());
-        assertEquals(1, executorService.getQueueSize());
+        CountDownLatch finishLatch = startLongRunningTask(executorService);
+        try {
+            executeNopTask(executorService);
+            assertEquals(1, executorService.getQueueSize());
+            assertEquals(1, executorService.getQueueSize());
+        } finally {
+            finishLatch.countDown();
+        }
     }
 
     @Test
@@ -146,10 +149,13 @@ public class CachedExecutorServiceDelegateTest {
         int queueSize = 10;
         ManagedExecutorService executorService = newManagedExecutorService(1, queueSize);
 
-        startInfinitelyRunningTask(executorService);
-        executeNopTask(executorService);
-
-        assertEquals(queueSize - 1, executorService.getRemainingQueueCapacity());
+        CountDownLatch finishLatch = startLongRunningTask(executorService);
+        try {
+            executeNopTask(executorService);
+            assertEquals(queueSize - 1, executorService.getRemainingQueueCapacity());
+        } finally {
+            finishLatch.countDown();
+        }
     }
 
     @Test
@@ -279,22 +285,26 @@ public class CachedExecutorServiceDelegateTest {
     @Test
     public void shutdownNow() throws Exception {
         ManagedExecutorService executorService = newManagedExecutorService();
-        startInfinitelyRunningTask(executorService);
-
-        Future<Object> future = executorService.submit(new Callable<Object>() {
-            @Override
-            public Object call() throws Exception {
-                return null;
-            }
-        });
-
-        List<Runnable> tasks = executorService.shutdownNow();
-        assertTrue(executorService.isShutdown());
-
-        assertEquals(1, tasks.size());
+        CountDownLatch finishLatch = startLongRunningTask(executorService);
         try {
-            future.get();
-        } catch (CancellationException expected) {
+
+            Future<Object> future = executorService.submit(new Callable<Object>() {
+                @Override
+                public Object call() throws Exception {
+                    return null;
+                }
+            });
+
+            List<Runnable> tasks = executorService.shutdownNow();
+            assertTrue(executorService.isShutdown());
+
+            assertEquals(1, tasks.size());
+            try {
+                future.get();
+            } catch (CancellationException expected) {
+            }
+        } finally {
+            finishLatch.countDown();
         }
     }
 
@@ -375,17 +385,18 @@ public class CachedExecutorServiceDelegateTest {
         });
     }
 
-    private void startInfinitelyRunningTask(ExecutorService executorService) {
-        final CountDownLatch latch = new CountDownLatch(1);
+    private CountDownLatch startLongRunningTask(ExecutorService executorService) {
+        final CountDownLatch startLatch = new CountDownLatch(1);
+        final CountDownLatch finishLatch = new CountDownLatch(1);
         executorService.execute(new Runnable() {
             @Override
             public void run() {
-                latch.countDown();
-                LockSupport.park();
+                startLatch.countDown();
+                assertOpenEventually(finishLatch);
             }
         });
-
-        assertOpenEventually(latch);
+        assertOpenEventually(startLatch);
+        return finishLatch;
     }
 
     private void executeNopTask(ExecutorService executorService) {

